src/charactersheet/models/common/image.js

```diff
@@ -23,28 +24,52 @@ export class EncounterImage extends KOModel {
 
     DESCRIPTION_MAX_LENGTH = 100;
 
-    // Public Methods
-
-    toJSON = function() {
-        return {
-            name: this.name(),
-            url: this.sourceUrl(),
-            description: this.playerText()
-        };
-    };
-
     shortDescription = ko.pureComputed(() => {
         return Utility.string.truncateStringAtLength(this.description(), this.DESCRIPTION_MAX_LENGTH);
     });
 
-    /* Message Methods */
+    convertedDisplayUrl = ko.pureComputed(() => (
+        this.sourceUrl()
+            ? Utility.string.createDirectDropboxLink(this.sourceUrl())
+            : null
+    ));
 
-    toHTML = function() {
-        return 'New image in chat';
-    };
+    load = async (params) => {
+        const response = await this.ps.model.ps.read(params);
+        this.importValues(response.object.exportValues());
+    }
+
+    create = async () => {
+        const response = await this.ps.create();
+        this.importValues(response.object.exportValues());
+        Notifications.encounterimage.added.dispatch(this);
+    }
+
+    save = async () => {
+        const response = await this.ps.save();
+        this.importValues(response.object.exportValues());
+        Notifications.encounterimage.changed.dispatch(this);
+    }
+
+    delete = async () => {
+        await this.ps.delete();
+        Notifications.encounterimage.deleted.dispatch(this);
+    }
 }
 
+
 EncounterImage.validationConstraints = {
+    fieldParams: {
```

Comment:
what is this used for? I forgot

Reply:
The HTML field validation. The current method spreads these attributes into the KO attribute binding. Any valid HTML attr is valid there.

#### Example

```
// Model
Encounter.validationConstraints = {
  fieldParams: {
    name: {
      required: true,
      maxlength: 200,
    }
  }
}

// Template
<input 
  data-bind="attr: { ...Encounter.validationConstraints.fieldParams.name }"
  name="name"
  other-attr="etc"
/>
```